File: 1627-last-moment-before-all-ants-fall-out-of-a-plank/1627-last-moment-before-all-ants-fall-out-of-a-plank.py
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        if (not left): return n - min(right)
        if (not right): return max(left)
        val1 = max(left)
        val2 = n - min(right)
        return max(val1, val2)

File: 1627-last-moment-before-all-ants-fall-out-of-a-plank/1627-last-moment-before-all-ants-fall-out-of-a-plank.java
class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        int time = 0;

        for (int l: left){
            time = Math.max(time, l);
        }

        for (int r: right){
            time = Math.max(time, n-r);
        }

        return time;
    }
}